src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -2425,6 +2580,54 @@ void AddType(INamedTypeSymbol type)
             }
         }
 
+        void AddCustomAttributes(TypeDeclaration typeDeclaration, string interfaceName = null)

```

Comment:
given any thought to reversing SupportedOSPlatform?  It could always map to UniversalApiContract of the appropriate nearest version (for which the OS version is less than or equal to the one supplied with SupportedOSPlatform).  Then, other language projections could based their own downlevel version checks on the Contract (a C#/WinRT does).

Reply:
I thought the contract attributes were only meant for Windows APIs as per [this](https://docs.microsoft.com/en-us/uwp/midl-3/predefined-attributes#the-contract-attribute).

Reply:
I suppose so - doesn't sound like there's any way then to enable a component author to assert an OS requirement

Reply:
I wonder if we can detect the target framework used and any SupportedOSPlatform and then when we generate our targets file which they include, we add something that checks if the targetplatformversion in the consumer is compatible with that.  I will open a issue to track investigating that.